torch/utils/data/dataloader.py

```diff
@@ -146,7 +146,10 @@ def pin_memory_batch(batch):
     elif isinstance(batch, collections.Mapping):
         return {k: pin_memory_batch(sample) for k, sample in batch.items()}
     elif isinstance(batch, collections.Sequence):
-        return [pin_memory_batch(sample) for sample in batch]
+        try:
+            return batch._make([pin_memory_batch(sample) for sample in batch])
```

Comment:
It would be better to do something like this: `type(batch)(*(pin_memory_batch(sample) for sample in batch))`.

Reply:
Okay, fixed it.